src/Accordion/AccordionSection.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import AccordionHeader from './AccordionHeader';
+import AccordionPanel from './AccordionPanel';
+
+class AccordionSection extends React.Component {
+  state = {
+    isOpen: false
+  };
+
+  get isControlled() {
+    const { isOpen } = this.props;
+
+    return isOpen !== undefined;
+  }
+
+  get isOpen() {
+    const { isOpen: controlledIsOpen } = this.props;
+    const { isOpen: internalIsOpen } = this.state;
+
+    return this.isControlled ? controlledIsOpen : internalIsOpen;
+  }
+
+  onHeaderClick = () => {
+    const { onHeaderClick } = this.props;
+
+    if (!this.isControlled) {
+      this.toggle();
+    }
+
+    onHeaderClick();
+  };
+
+  toggle() {
+    const { isOpen } = this.state;
+    this.setState({ isOpen: !isOpen });
```

Comment:
`this.setState(prevstate => ({ isOpen: !prevState.isOpen }))` for React `setState` batching.